libs/core/src/main/java/org/opensearch/core/tasks/TaskId.java

```diff
@@ -84,9 +91,11 @@ public TaskId(String taskId) {
             } catch (NumberFormatException ex) {
                 throw new IllegalArgumentException("malformed task id " + taskId, ex);
             }
+            taskIdProto = TaskIdProto.TaskId.newBuilder().setNodeId(nodeId).setId(id).build();
         } else {
             nodeId = "";
             id = -1L;
+            taskIdProto = TaskIdProto.TaskId.newBuilder().setNodeId(nodeId).setId(id).build();
```

Comment:
See above on serializing "empty" information vs. never even looking at this object.